### ISSUE TITLE
IPL: cckddasd.c:1760: rc=22: Invalid argument

### ISSUE THREAD

**Issue by GarethJacobsNZ** [SDL-Hercules-390/hyperion#294]
During IPL the following error message is displayed after BSPSETF ENDED...

```
/  BSPSP21I - Member being processed: SETPFK01                                                            
/ +BSPSP98I - Member processed, LASTCC=0000                                                              
/ +BSPSP21I - Member being processed: SETPFK02                                                           
/ +BSPSP98I - Member processed, LASTCC=0000                                                              
/ +BSPSP99I - End of processing, MAXRC=0000                                                              
/ $HASP395 BSPSETPF ENDED                                                                                
HHC00107I Starting thread cckd_writer() from cckd_writer(), active=1, started=1, max=2                   
HHC90020W 'hthread_setschedparam()' failed at loc=cckddasd.c:1760: rc=22: Invalid argument               
HHC00007I Previous message from function 'hthread_set_thread_prio' at hthreads.c(1162)                   
HHC00100I Thread id 00007f28209dc700, prio 0, name 'cckd_writer thread 2' started
```

Operatng system: **Linux-5.3.0-46-generic x86_64**
Emulation: **MVS 3.8J Update 08**
Hercules version:
```
HHC01413I Hercules version 4.3.9999.0-SDL-g017e4374-modified (4.3.9999.0)
HHC01414I (C) Copyright 1999-2020 by Roger Bowler, Jan Jaeger, and others
HHC01417I ** The SoftDevLabs version of Hercules **
HHC01415I Build date: Apr 20 2020 at 14:51:16
HHC01417I Built with: GCC 10.0.1 20200416 (experimental) [master revision 3c3f12e2a76:dcee354ce56:44b326839d864fc10c459916abcc97f35a9ac3de]
HHC01417I Build type: GNU/Linux x86_64 host architecture build
HHC01417I Modes: S/370 ESA/390 z/Arch
HHC01417I Max CPU Engines: 128
```

Although I built this with GCC 10, the same error occurred when built under GCC 9.

The error doesn't appear to cause any problems post-IPL. TSO Login and functionality seems to be normal.

Please let me know if you need further.

Thanks
Gareth

**Comment by mcisho** [comment]
Are you running Hercules without privileges? I am, and I see around 10 of these HHC90020W messages from various threads, along with several other warning and error messages related to priorities et al. The GCC version is irrelevant.
Ian

**Comment by GarethJacobsNZ** [comment]
Yes, runs unprivileged. Don't recall seeing those messages in previous versions of Hercules, but if you are getting the same messages (and more) then probably okay.
I re-compiled with GCC 9 to eliminate the compiler as the cause as GCC 10 is experimental - so yes irrelevant.

**Comment by Fish-Git** [comment]
> Yes, runs unprivileged.

That's your problem: setting priorities requires elevated privileges. The messages are considered normal when running Hercules unprivileged.

The default when building Hercules is `--enable-capabilities`, so unless you specifically disabled that by specifying `--disable-capabilities` on your ./configure command, Hercules is built with capabilities support, thereby allowing you run Hercules with root privileges -- _when needed_.

That is to say, Hercules is started as root, but then immediately _drops_ root privileges so that 99.99% of the time it runs, it runs with normal _non-root_ privileges, and only reacquires root privileges when it needs to in order to do something requiring it (such as setting thread priorities for example), so it should be safe. That is to say, starting Hercules elevated (as root? I'm not a Linux person!) _should_ be safe. Doing so should _not_ (as far as I know!) introduce any type of security compromise/exposure.

So, bottom line: the warnings you are seeing are normal. Just ignore them.

OR... run Hercules privileged (thereby allowing Hercules to set its internal thread priorities as it sees fit) and you won't see them.

Your choice.

Am going to close this issue as "Invalid" (user error) and "Won't Fix" (working as designed).
